src/test/scala/com/databricks/spark/redshift/RedshiftSuite.scala

```diff
@@ -0,0 +1,458 @@
+      1.0f, 42, 1239012341823719L, 23.toShort, "Unicode's樂趣", TestUtils.toTimestamp(2015, 6, 1, 0, 0, 0, 1))
+  )
+
+
+  /**
+   * Spark Context with hadoop file overridden to point at our local test data file for this suite,
+   * no-matter what temp directory was generated and requested.
+   */
+  private var sc: SparkContext = _
+  private var conn: Connection = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sc = new SparkContext("local", "RedshiftSourceSuite")
+
+    sc.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", accessKeyId)
+    sc.hadoopConfiguration.set("fs.s3n.awsSecretAccessKey", secretAccessKey)
+
+    conn = DefaultJDBCWrapper.getConnector("org.postgresql.Driver", jdbcUrl, new Properties())()
+
+    conn.prepareStatement("drop table if exists test_table").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table2").executeUpdate()
+    conn.prepareStatement("drop table if exists test_table3").executeUpdate()
+    conn.commit()
+
+    def createTable(tableName: String): Unit = {
+      conn.prepareStatement(
+        s"""
+           |create table $tableName (
+           |testbyte int2,
+           |testbool boolean,
+           |testdate date,
+           |testdouble float8,
+           |testfloat float4,
+           |testint int4,
+           |testlong int8,
+           |testshort int2,
+           |teststring varchar(256),
+           |testtimestamp timestamp
+           |)
+      """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |null, null, null, null, null, null, null, null, null, null)
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |0, null, '2015-07-03', 0.0, -1.0, 4141214, 1239012341823719, null, 'f', '2015-07-03 00:00:00.000')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |0, false, null, -1234152.12312498, 100000.0, null, 1239012341823719, 24, '___|_123', null)
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |1, false, '2015-07-02', 0.0, 0.0, 42, 1239012341823719, -13, 'asdf', '2015-07-02 00:00:00.000')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.prepareStatement(
+        s"""
+           |insert into $tableName values (
+           |1, true, '2015-07-01', 1234152.12312498, 1.0, 42,
+           |1239012341823719, 23, 'Unicode''s樂趣', '2015-07-01 00:00:00.001')
+        """.stripMargin
+      ).executeUpdate()
+
+      conn.commit()
+    }
+
+    createTable("test_table")
+    createTable("test_table2")
+    createTable("test_table3")
+
+    conn.commit()
+  }
+
+  override def afterAll(): Unit = {
+    val temp = new File(tempDir)
```

Comment:
Isn't `tempDir` an `s3://...` URI? If so, this won't be able to delete files on S3.

I think that we should be using a Hadoop filesystem instead.